### PR TITLE
Pre publish pages

### PR DESCRIPTION
**Pull request by awais786**
- Contains pre-publisher pages.
- Sass file.
- Test cases

screen shot 2016-07-31 at 5 06 23 pm

### CODE REVIEW

**Comment by awais786** [comment]
@bderusha Earlier I have created the branch from bderusha/course-detail but that branch is not merged yet. So I have created the PR from master. I will complete test cases today. I have attached the screenshot also.

File: course_discovery/templates/publisher/pre_publish/home.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{% block stylesheets %}
+    <link rel="stylesheet" href="{% static 'sass/publisher/pre_publish.scss' %}" type="text/x-scss">
+{% endblock %}
+
+{% block content %}
+
+<nav class="administration-navbar">
+    <div class="container">
+
+        <a class="selected" href="#tab-1">All</a>
```
Comment (bderusha) [inline course_discovery/templates/publisher/pre_publish/home.html]: i18n using `trans`

Comment (bderusha) [inline course_discovery/templates/publisher/pre_publish/home.html]: for this file and the other

File: course_discovery/templates/publisher/pre_publish/_studio.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+{% endblock title %}
+
+{% block content %}
+    <div class="course-information">
+        <div class="info-item">
+            <div class="heading">
+                Course Name
+            </div>
+
+            <div class="course-id">{{ object.wrapped_obj.course.title }}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/pre_publish/_studio.html]: expose this in the wrapper

File: course_discovery/templates/publisher/pre_publish/_studio.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        </div>
+        <div class="info-item">
+            <div class="heading">Organization</div>
+            <div class="course-id">
+                {% if object.course.organizations.all %}
+                    {{ object.course.organizations.all.0.key }}
```
Comment (bderusha) [inline course_discovery/templates/publisher/pre_publish/_studio.html]: expose this in the wrapper

File: course_discovery/templates/publisher/pre_publish/_studio.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                {% else %}
+                    N/A
+                {% endif %}
+            </div>
+        </div>
+        <div class="info-item">
+            <div class="heading">Number</div>
+            <div class="course-id">{{  object.course.number }}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/pre_publish/_studio.html]: expose this in the wrapper

File: course_discovery/templates/publisher/pre_publish/home.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        </div>
+        <div id="tab-4" class="tab-content">
+            {% include 'publisher/pre_publish/_drupal.html' %}
+        </div>
+    </div>
+</div>
+{% endblock %}
+
+{% block javascript %}
+    <script src="{% static 'js/publisher/publishers.js' %}"></script>
```
Comment (bderusha) [inline course_discovery/templates/publisher/pre_publish/home.html]: where is this defined?

**Comment by bderusha** [comment]
Overall this is a good start.  I have a few general comments:
- naming.  Based on the designs Roderick came up with what we've been calling the `pre-publish` page is actually just the `course run details` page, expanded to fit this publishing case.  We should use standard detail page naming conventions throughout the app.
- The css.  It looks good, but without designs from roderick I am wondering where all of this custom css came from?

**Comment by awais786** [comment]
@bderusha I have rename the page. And for css I have added some css for tabs and some custom css for better look. Once i receive the final design from roderick I will update it accordingly.

File: bower.json [hunk 6]
```diff
@@ -13,6 +13,8 @@
     "edx-pattern-library": "0.16.3",
     "underscore": "~1.8.3",
     "moment": "~2.10.3",
-    "pikaday": "https://github.com/owenmead/Pikaday.git#1.4.0"
+    "pikaday": "https://github.com/owenmead/Pikaday.git#1.4.0",
+    "jquery": "~2.1.4",
+    "requirejs": "~2.1.15"
```
Comment (waheedahmed) [inline bower.json]: requirejs is already in `package.json`.

**Comment by awais786** [comment]
@bderusha Please review this PR in detail.

File: course_discovery/static/sass/publisher/run_detail.scss [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+      padding: 10px;
+      color: #0ea6ec;
+
+      &.selected {
+        border-bottom: 4px solid #0ea6ec;
+        color: black;
+      }
+
+      &:first-child {
+        margin-left: 30px;
```
Comment (bderusha) [inline course_discovery/static/sass/publisher/run_detail.scss]: need `RTL` support

File: course_discovery/static/sass/publisher/run_detail.scss [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+
+    &.active {
+      color: #777777;
+    }
+  }
+}
+
+
+.copy-text {
+  margin-left: 5px;
```
Comment (bderusha) [inline course_discovery/static/sass/publisher/run_detail.scss]: needs `RTL` support

File: course_discovery/templates/publisher/run_detail/_cat.html [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+{% block title %}
+    {% trans "Course Run Detail" %}
+{% endblock title %}
+
+{% block content %}
+
+   <div class="course-information">
+        <div class="info-item">
+            <div class="heading">{% trans "Course ID" %}</div>
+            <div class="course-id">{{ object.number }}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_cat.html]: This is not the course id.  We need the `lms_course_id` here

File: course_discovery/templates/publisher/run_detail/_cat.html [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        </div>
+        <div class="info-item">
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_cat.html:14]: Pacing and dates are not required for CAT

File: course_discovery/templates/publisher/run_detail/_drupal.html [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+        <div>{{ object.number }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Course ID" %}</div>
+        <div>{{ object.lms_course_id }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Price" %}</div>
+        <div>
+            {{ object.verified_seat_price }}
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_drupal.html]: NIT: make this a 1-liner like the rest

File: course_discovery/templates/publisher/run_detail/_drupal.html [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+        </div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Sub Title" %}</div>
+        <div>{{ object.short_description }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "School" %}</div>
+        <div>
+            {% if object.organization %}
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_drupal.html]: handle this in the wrapper

File: course_discovery/templates/publisher/run_detail/_studio.html [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    <div class="course-information">
+        <div class="info-item">
+            <div class="heading">{% trans "Course Name" %}</div>
+
+            <div class="course-id">{{ object.title }}</div>
+        </div>
+        <div class="info-item">
+            <div class="heading">{% trans "Organization" %}</div>
+            <div class="course-id">
+                {{ object.organization.key }}
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_studio.html]: can currently return `None` which will error out with `NoneType has no method 'key'`

File: course_discovery/templates/publisher/run_detail/_studio.html [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+            </div>
+        </div>
+        <div class="info-item">
+            <div class="heading">{% trans "Number" %}</div>
+            <div class="course-id">{{  object.number}}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_studio.html]: NIT: add space at the end

**Comment by bderusha** [comment]
@awais786 Can you go through each tab with its counterpart open and confirm that we are only putting required data in our fields?

e.g   open CAT and our click the CAT tab on this page and make sure that the fields are all present, in the same order, and that we don't have extra fields on our end.

File: course_discovery/static/sass/publisher/run_detail.scss (hunk 7 above)
Comment (awais786) [inline course_discovery/static/sass/publisher/run_detail.scss]: Should i wait for the final css from Roderick or implement it by self ?

Comment (bderusha) [inline course_discovery/static/sass/publisher/run_detail.scss]: We need to get into the habit, so please add it

**Comment by awais786** [comment]
Is  this the drupal page for reference?
screen shot 2016-08-02 at 5 52 17 pm

File: course_discovery/templates/publisher/run_detail/_credit_seat.html [hunk 15]
```diff
@@ -0,0 +1,26 @@
+{% load i18n %}
+<h3 class="hd-6 example-set-hd">{% trans "Credit Seats" %}</h3>
```
Comment (waheedahmed) [inline course_discovery/templates/publisher/run_detail/_credit_seat.html]: Change class name `example-set-hd` to something else.

File: course_discovery/templates/publisher/run_detail/_credit_seat.html [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+{% if object.credit_seats %}
+<div class="example-set">
```
Comment (waheedahmed) [inline course_discovery/templates/publisher/run_detail/_credit_seat.html]: Change class name `example-set`.

File: course_discovery/templates/publisher/run_detail/_seats.html [hunk 17]
```diff
@@ -0,0 +1,24 @@
+{% load i18n %}
+<h3 class="hd-4 example-set-hd">Seats</h3>
```
Comment (waheedahmed) [inline course_discovery/templates/publisher/run_detail/_seats.html]: Change class name `example-set-hd`.

File: course_discovery/templates/publisher/run_detail/_seats.html [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+{% if object.non_credit_seats %}
+<div class="example-set">
```
Comment (waheedahmed) [inline course_discovery/templates/publisher/run_detail/_seats.html]: Change class name `example-set`.

File: course_discovery/static/sass/publisher/course_detail.scss [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+.tab-content {
+  display: none;
+
+  &.active {
+    display: block;
+  }
+}
+
+
+.example-set {
```
Comment (waheedahmed) [inline course_discovery/static/sass/publisher/course_detail.scss]: Change class name `example-set` to something else.

File: course_discovery/templates/publisher/run_detail/_credit_seat.html (hunk 15 above)
Comment (awais786) [inline course_discovery/templates/publisher/run_detail/_credit_seat.html]: I pick this from patter library. [ux](http://ux.edx.org/components/tables/)
Also this is temporary we will get final css from Roderick.

File: course_discovery/apps/publisher/wrappers.py [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+        return ", " .join(
+            [
+                self.wrapped_obj.course.primary_subject.name,
+                self.wrapped_obj.course.secondary_subject.name,
+                self.wrapped_obj.course.tertiary_subject.name
+            ]
+        )
+
+    @property
+    def course_type(self):
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/wrappers.py:88]: I think its seat type not course type.

**Comment by waheedahmed** [comment]
Few minor comments, 👍 once addressed.

File: course_discovery/apps/publisher/wrappers.py (hunk 20 above)
Comment (awais786) [inline course_discovery/apps/publisher/wrappers.py:88]: In cat it is mentioned as courst type. That's why I did it.

**Comment by awais786** [comment]
@bderusha Please review this PR.

File: course_discovery/static/sass/publisher/course_form.scss [hunk 21]
```diff
@@ -6,6 +6,7 @@
 }
 
 .btn-add {
-    float: right;
-    margin-right: 30px;
+  float: right;
+  margin-right: 30px;
```
Comment (bderusha) [inline course_discovery/static/sass/publisher/course_form.scss]: `RTL`

File: course_discovery/static/sass/publisher/course_form.scss [hunk 22]
```diff
@@ -6,6 +6,7 @@
 }
 
 .btn-add {
-    float: right;
-    margin-right: 30px;
+  float: right;
```
Comment (bderusha) [inline course_discovery/static/sass/publisher/course_form.scss]: `RTL`

File: course_discovery/templates/publisher/run_detail/_drupal.html [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+    <div class="info-item">
+        <div class="heading">{% trans "what you'll learn" %}</div>
+        <div>{{ object.expected_learnings }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Prerequisite" %}</div>
+        <div>{{ object.prerequisites }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "KeyWords" %}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_drupal.html]: Keywords

File: course_discovery/templates/publisher/run_detail/_drupal.html [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+    <div class="info-item">
+        <div class="heading">{% trans "Level" %}</div>
+        <div>{{ object.level_type }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "About this Course" %}</div>
+        <div>{{ object.full_description }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "what you'll learn" %}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_drupal.html]: `What`

File: course_discovery/templates/publisher/run_detail/_drupal.html [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+        <div class="heading">{% trans "Self Paced" %}</div>
+        <div>{{ object.pacing_type }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Staff" %}</div>
+        <div>{{ object.persons }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Estimated Effort" %}</div>
+        <div>{{ object.min_effort }} to {{ object.max_effort }} week</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_drupal.html]: hours per week

File: course_discovery/templates/publisher/run_detail/_drupal.html [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+        <div class="heading">{% trans "Prerequisite" %}</div>
+        <div>{{ object.prerequisites }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "KeyWords" %}</div>
+        <div>{{ object.wrapped_obj.keywords }}</div>
+    </div>
+    <div class="info-item">
+        <div class="heading">{% trans "Sponsors" %}</div>
+        <div>{{ object.partner }}</div>
```
Comment (bderusha) [inline course_discovery/templates/publisher/run_detail/_drupal.html]: Sponsors and partners are different.  CourseRuns have a `sponsors` attr.  Please use that

File: course_discovery/apps/publisher/views.py [hunk 27]
```diff
@@ -122,3 +123,14 @@ def form_valid(self, form):
 
     def get_success_url(self):
         return reverse(self.success_url, kwargs={'pk': self.object.id})
+
+
+class CourseRunDetailView(DetailView):
+    """ Pre Publish View."""
```
Comment (bderusha) [inline course_discovery/apps/publisher/views.py]: Update docstring

File: course_discovery/apps/publisher/urls.py [hunk 28]
```diff
@@ -13,5 +13,5 @@
     url(r'^course_runs/(?P<pk>\d+)/edit/$', views.UpdateCourseRunView.as_view(), name='publisher_course_runs_edit'),
     url(r'^seats/new$', views.CreateSeatView.as_view(), name='publisher_seats_new'),
     url(r'^seats/(?P<pk>\d+)/edit/$', views.UpdateSeatView.as_view(), name='publisher_seats_edit'),
-
+    url(r'^course_run_detail/(?P<pk>\d+)/$', views.CourseRunDetailView.as_view(), name='course_run_detail'),
```
Comment (bderusha) [inline course_discovery/apps/publisher/urls.py]: `url(r'^course_runs/(?P<pk>\d+)/$', views.CourseRunDetailView.as_view(), name='publisher_course_run_detail'),`

Comment (bderusha) [inline course_discovery/apps/publisher/urls.py]: please group with other `course_runs` urls

File: course_discovery/apps/publisher/tests/test_wrapper.py [hunk 29]
```diff
@@ -49,3 +53,50 @@ def test_callable(self):
...
+        ([], Seat.AUDIT),
+        ([Seat.AUDIT], Seat.AUDIT),
+        ([Seat.AUDIT, Seat.CREDIT, Seat.VERIFIED], Seat.CREDIT),
+        ([Seat.AUDIT, Seat.VERIFIED], Seat.VERIFIED),
+        ([Seat.PROFESSIONAL], Seat.PROFESSIONAL),
+    )
+    @ddt.unpack
+    def test_course_type(self, seats, course_type):
+        self._generate_seats(seats)
+        wrapper_object = CourseRunWrapper(self.course_run)
```
Comment (bderusha) [inline course_discovery/apps/publisher/tests/test_wrapper.py:66]: use `self.wrapped_course_run` here and throughout

**Comment by bderusha** [comment]
Some comments and quality errors.

👍 once addressed and green build

**Pull request merged**